### PR TITLE
Fix panics when subcommand is missing

### PR DESCRIPTION
**Pull request by BenCookie95**
### Ticket
https://github.com/mattermost-community/mattermost-plugin-google-drive/issues/41

### Test cases
Use the ticket description for testing

Also worth doing a quick sanity test on receiving notifications for comments, deleted comments and resolved commets. Make sure the received message is formatted correctly as before

### CODE REVIEW

**Review by davidkrauser — COMMENTED** [review]
Looks good. I took a quick search through the codebase and found a few more similar cases where we are accessing by index, but not checking the size of the slice:

https://github.com/mattermost-community/mattermost-plugin-google-drive/blob/269439f5076f7051a5428845583700e0f098b8d9/server/plugin/notifications.go#L118

https://github.com/mattermost-community/mattermost-plugin-google-drive/blob/269439f5076f7051a5428845583700e0f098b8d9/server/plugin/notifications.go#L147

https://github.com/mattermost-community/mattermost-plugin-google-drive/blob/269439f5076f7051a5428845583700e0f098b8d9/server/plugin/notifications.go#L153

https://github.com/mattermost-community/mattermost-plugin-google-drive/blob/269439f5076f7051a5428845583700e0f098b8d9/server/plugin/create.go#L41-L42

I don't know if these are issues, but I thought I'd point them out. If they are, might be a good opportunity to knock these out, too :slightly_smiling_face:

**Comment by BenCookie95** [comment]
> Looks good. I took a quick search through the codebase and found a few more similar cases where we are accessing by index, but not checking the size of the slice:
> 
> https://github.com/mattermost-community/mattermost-plugin-google-drive/blob/269439f5076f7051a5428845583700e0f098b8d9/server/plugin/notifications.go#L118
> 
> https://github.com/mattermost-community/mattermost-plugin-google-drive/blob/269439f5076f7051a5428845583700e0f098b8d9/server/plugin/notifications.go#L147
> 
> https://github.com/mattermost-community/mattermost-plugin-google-drive/blob/269439f5076f7051a5428845583700e0f098b8d9/server/plugin/notifications.go#L153
> 
> https://github.com/mattermost-community/mattermost-plugin-google-drive/blob/269439f5076f7051a5428845583700e0f098b8d9/server/plugin/create.go#L41-L42
> 
> I don't know if these are issues, but I thought I'd point them out. If they are, might be a good opportunity to knock these out, too 🙂

Thanks for finding those, those responses are coming from Google but it's worth double checking their indexes exist. I'll push a fix for those in a sec

File: server/plugin/create.go [hunk 1]
```diff
@@ -173,6 +173,10 @@ func appendUsersWithoutAccessSlice(config *model.Config, usersWithoutAccess []st
 }
 
 func (p *Plugin) handleCreate(c *plugin.Context, args *model.CommandArgs, parameters []string) string {
```
[the remark this answers is not in the document]
Comment (BenCookie95) [reply server/plugin/create.go:175]: Good question, ideally these would return the standard error type. I think the pattern was copied over from some older plugins such as this https://github.com/mattermost/mattermost-plugin-github/blob/827e88111cce39a19892be6b9363658643e85bc2/server/plugin/command.go#L859.

**Comment by BenCookie95** [comment]
@davidkrauser I think I covered any potential panics, any function that calls `getCommentUsingDiscussionID` is already protected from a panic so I didn't add an extra check to those

**Review by fmartingr — APPROVED** [review]

File: server/plugin/create.go [hunk 2]
```diff
@@ -32,14 +32,27 @@ func (p *Plugin) sendFileCreatedMessage(ctx context.Context, channelID, fileID,
...
+		userPhotoLink := ""
+		if len(file.Owners) != 0 {
+			userDisplayName = file.Owners[0].DisplayName
+			userPhotoLink = file.Owners[0].PhotoLink
+		} else {
+			user, err := p.Client.User.Get(userID)
+			if err != nil {
+				return errors.Wrap(err, "failed to fetch user")
+			}
+			userDisplayName = user.Username
```
Comment (davidkrauser) [inline server/plugin/create.go:45]: **question:** are we able to set the photo link here? Is it okay to leave it an empty string?

File: server/plugin/notifications.go [hunk 3]
```diff
@@ -67,15 +67,18 @@ func (p *Plugin) handleAddedComment(ctx context.Context, dSrv google.DriveInterf
 }
 
 func (p *Plugin) handleDeletedComment(userID string, activity *driveactivity.DriveActivity, file *drive.File) {
-	urlToComment := activity.Targets[0].FileComment.LinkToDiscussion
-	message := fmt.Sprintf("A comment was deleted in %s %s", utils.GetInlineImage("Google failed:", file.IconLink), utils.GetHyperlink(file.Name, urlToComment))
+	urlToComment := file.Name
```
Comment (davidkrauser) [inline server/plugin/notifications.go:70]: **question:** if we can't resolve a `LinkToDiscussion`, we'll use `file.Name` instead. What's the difference between the two?

**Review by davidkrauser — APPROVED** [review]
Pre-approving, just have a few questions for my own edification :-)

File: server/plugin/create.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidkrauser) [reply server/plugin/create.go:188]: Ahh, thanks for the link. If I understand correctly, it looks like we have functions that could potentially return strings to be displayed to the user, and if the string is nil we don't display anything. For this code it just happens that all of the strings are errors, but we could have other functions that return non-error messages.

File: server/plugin/notifications.go (hunk 3 above)
Comment (BenCookie95) [reply server/plugin/notifications.go:70]: `LinkToDiscussion` will link directly to the conversation this activity is referencing but `file.Name` will at least tell them what file this comment is from. Now that you've pointed this out though I think I'll also make this a direct link to the file as well.

**Review by arush-vashishtha — APPROVED** [review]
I have tested the slash commands without arguments and the plugin is not panicking anymore. LGTM
Also I saw that you have added a new feature to show the user's name and the user image who has created the document. I looked around it a little and it is working fine, but maybe like do want me to test some more edge cases around it? If so please do let me know,
image

**Comment by BenCookie95** [comment]
> I have tested the slash commands without arguments and the plugin is not panicking anymore. LGTM Also I saw that you have added a new feature to show the user's name and the user image who has created the document. I looked around it a little and it is working fine, but maybe like do want me to test some more edge cases around it? If so please do let me know, image

Thanks for checking Arush, I don't think it's possible to manually test that because Google would have to omit some fields from it's response in order to trigger that code path
